nservicebus/architecture/publishing-from-web-applications.md

```diff
@@ -0,0 +1,89 @@
+
+Yes, it is possible, and in some circumstances, even advisable to publish events from within a web application. This article will describe the guidelines for publishing messages within web applications under different circumstances.
+
+## Why/Why Not?
+
+Historically, there have been good reasons to avoid publishing events from within web applications:
+
+1. **Transactions and Consistency** - Generally good events should announce that something has *already* happened. Many times this involves making some changes to a database and then publishing the result. But HTTP is inherently unreliable and does not have built-in retries. If an exception occurs before the event is published, you may lose your only opportunity to publish that event. In these circumstances, it's generally better to send a command with the payload of the HTTP request, and have another endpoint process that command with the advantages of automatic retry and an error queue.
+2. **Web Application Scale-Out** - Because events can only be published from one logical publisher, it can be problematic to scale out a web application that is publishing events. More on this later.
+3. **Web App Recycling** - Prior to IIS 7.5 introducing the AlwaysRunning feature (AKA AlwaysOn), web applications not receiving could be recycled or idled at any time, which would result in incoming messages building up in an input queue, without an application actively processing them.
+
+Because of this, conventional wisdom has suggested that when in the context of a web application, it is better to only send commands to a back-end service endpoint, which can then publish a similar event.
+
+But, things change:
+
+* With IIS 7.5 and later, IIS supports AlwaysRunning mode.
+* IIS is no longer the only way to host web applications, which can now be hosted by [OWIN](http://owin.org/)/[Katana](http://www.asp.net/aspnet/overview/owin-and-katana), [NancyFx](http://nancyfx.org/), [FubuMVC](http://mvc.fubu-project.org/), and others. This largely removes the concern over web application recycling.
+* NServiceBus 4.0 introduced the ability to have transports besides MSMQ. Not all of the transports handle Publish/Subscribe in the same way as MSMQ - some even support it natively.
+* Cloud transports generally have an associated cost per interaction. This means that requiring an unnecessary send before publish costs extra money to operate.
+* In NServiceBus 4.0 and 5.0, the process of creating a send-only endpoint (which is still capable of publishing messages, but does not have an input queue or process incoming messages) was made much easier.
+
+## Publish/Subscribe Mechanics
+
+First it's important to understand how publish/subscribe actually works, consisting of the subscription phase, and then the actual publish operation.

```

Comment:
We have some doco on this that we should be able to link to eg. http://docs.particular.net/nservicebus/messaging/publish-subscribe/